src/main/java/io/mosip/mimoto/controller/IdpController.java

```diff
@@ -96,45 +107,38 @@ public ResponseEntity<Object> request(@RequestBody WalletBindingRequestDTO reque
             if (internalResponse == null)
                 throw new IdpException();
 
-            response = joseUtil.addThumbprintAndKeyId(internalResponse);
-            return ResponseEntity.status(HttpStatus.OK).body(response);
+            responseWrapper = joseUtil.addThumbprintAndKeyId(internalResponse);
+            return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
         } catch (Exception e) {
-            logger.error("Wallet binding error occured for tranaction id " + requestDTO.getRequest().getIndividualId(), e);
-            ResponseWrapper<Object> response = getErrorResponse(PlatformErrorMessages.MIMOTO_WALLET_BINDING_EXCEPTION.getCode(), e.getMessage());
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(response);
+            log.error("Wallet binding error occured for tranaction id " + requestDTO.getRequest().getIndividualId(), e);
+            String[] errorObj = Utilities.handleExceptionWithErrorCode(e);
+            List<ErrorDTO> errors = Utilities.getErrors(errorObj[0], errorObj[1]);
+            responseWrapper.setResponse(null);
+            responseWrapper.setErrors(errors);
```

Comment:
as we know api response can have either error or response then shouldn't we override the setters for response wrappers as
```
setResponse(response) {
this.response = response;
this.errors = null;
}
setErrors(errors) {
this.response = null;
this.errors = errors;
}
```

so that developers don't miss setting them null explicitly

Reply:
Added the Setters for Response